### PR TITLE
FOIA-238: Updates documentation for FOIA Upload XML.

### PR DESCRIPTION
**Pull request by alozie**

### CODE REVIEW

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 1]
```diff
@@ -1,61 +1,121 @@
-# Import data from agency annual reports in NIEM-XML format
-
-This module has the following components:
+#Table of Contents
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Need space after # and a line break after.

Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: Also should be an h2

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 2]
```diff
@@ -1,61 +1,121 @@
...
+ * Installation
+ * Usage
+ * Configuration
+ * Troubleshooting
+ * FAQ
+
+
+INTRODUCTION
+------------
+The FOIA Upload XML module enables agencies to upload annual reports to the
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 3]
```diff
@@ -1,61 +1,121 @@
...
+The FOIA Upload XML module enables agencies to upload annual reports to the 
+FOIA site via NIEM-XML formatted annual report documents.
+
+The module creates a new Annual Report node per agency and report year or 
+updates an already existing report for the corresponding agency and report year.
+
+
+REQUIREMENTS
+------------
+This module requires the core Migrate module along with the following contrib
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 4]
```diff
@@ -1,61 +1,121 @@
...
+ * Migrate Plus (https://www.drupal.org/project/migrate_plus)
+ * Migrate Tools (https://www.drupal.org/project/migrate_tools)
+ 
+For development, maintenance, and troubleshooting installing Drush is strongly 
+recommended.
+
+
+INSTALLATION
+-------------
+ * FOIA Upload XML is a custom Drupal module so unlike contrib modules, it is
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 5]
```diff
@@ -1,61 +1,121 @@
...
+ * Enable the foia_upload_xml module as you would normally install 
+ modules to your platform. 
+ 
+ * For more information about module installation in Drupal 8 see: 
+ https://www.drupal.org/docs/8/extending-drupal-8/installing-drupal-8-modules
+
+
+USAGE
+-----
+Once installed, users with `create annual_foia_report_data content` permission
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break between before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 6]
```diff
@@ -1,61 +1,121 @@
...
+
+Should the report format be updated the following section will describe how the 
+report migration mappings are currently setup and how new mappings might be 
+added should new report sections and/or fields be added.
+
+
+# Updating and Adding Report Sections
+
+## Overview
+This module defines multiple migrations for importing a full annual report. It
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 7]
```diff
@@ -1,61 +1,121 @@
...
+progress a message indicating `Your report has been queued...` will appear. 
+
+Once complete navigate to the Content section to find new or updated report. 
+Should any errors or other issues occur with the upload, users should contact 
+their Agency Managers or Agency Administrators.
+
+
+CONFIGURATION
+-------------
+# Contents
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 8]
```diff
@@ -1,61 +1,121 @@
...
+ * Post-Installation Configuration
+ * Updating and Adding Report Sections
+
+# Post-Installation Configuration
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add another line break before (2 breaks between sections).

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 9]
```diff
@@ -1,61 +1,121 @@
...
-
-```
-drush mim --group=foia_xml --update
-```
-
-Use `drush ms --group=foia_xml` to check the status of the migrations. To see
-available options, use `drush help mim`.
-
-## Custom migrations
+### Custom migrations
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before (2 breaks between sections).

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 10]
```diff
@@ -1,61 +1,121 @@
...
+
 
-The file `migrate_plus.migration_group.foia_xml.yml` defines the `foia_xml`
-migration group. All the custom migrations belong to this group.
+#### Beginning and end
 
-The file `migrate_plus.migration.foia_agency_report.yml` creates a single
-`annual_foia_report_data` node from the XML report. It depends on all the
-other migrations, so it will only run when they are complete.
+##### Main Report Migration - `foia_agency_report`
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before (2 breaks between sections).

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 11]
```diff
@@ -1,61 +1,121 @@
...
+The file `migrate_plus.migration.foia_agency_report.yml` defines the main report
+import and creates a single `annual_foia_report_data` node from the XML report. 
+It depends on all the other migrations, so it will only run when they are 
+complete.
 
+##### Agency Component Mapping - `component`
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before (2 breaks between sections).

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 10 above)
Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: Use sentence casing.

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 11 above)
Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: Use sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 7 above)
Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: make h2

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 12]
```diff
@@ -1,61 +1,121 @@
...
+`create annual_foia_report_data content` permission will be able to upload XML 
+report files for their assigned agency. No further configuration will be 
+required.
+
+Should the report format be updated the following section will describe how the 
+report migration mappings are currently setup and how new mappings might be 
+added should new report sections and/or fields be added.
+
+
+# Updating and Adding Report Sections
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: should be h2

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 6 above)
Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: should be h3

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 12 above)
Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: Fix headings throughout document based on this please

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 13]
```diff
@@ -1,61 +1,121 @@
...
-
-## Custom migrations
+### Custom migrations
 
 There are a lot of files in the `/config/install/` directory, but it is not
 too complicated once you understand how they are organized.
 
-### Beginning and end
+By group they are:
+- `foia_xml`: Main migrations - annual report node (`foia_agency_report`) with
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: For bullets there should be a space before then an asterisk not dash.
[space][asterisk][space][text]

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 14]
```diff
@@ -85,11 +145,12 @@ report year, agency abbreviation, internal identifier to the node ID of the
 corresponding `agency_component` node. The first two keys are used to identify
 the current XML report, or the corresponding `annual_foia_report_data` node.
 
-### The middle
+#### The middle
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 15]
```diff
@@ -85,11 +145,12 @@ report year, agency abbreviation, internal identifier to the node ID of the
...
 
 For each paragraph field (and associated "overall" fields) we add two
 migrations.
 
+##### Component Data Mapping
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before

Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: use sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 16]
```diff
@@ -115,18 +176,19 @@ handled by the `component` migration. The file
 section-specific identifiers "PS0" and "PS19" and so on to the corresponding
 `agency_component` nodes.
 
+##### Component Data Import
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before

Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: use sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 17]
```diff
@@ -115,18 +176,19 @@ handled by the `component` migration. The file
...
 The second migration for this section is defined by the file
 `migrate_plus.migration.foia_requests_va.yml`. This migration creates
 Paragraphs of type `foia_req_va`. These Paragraphs are then referenced in
 `field_foia_requests_va` in the `foia_agency_report` migration, using the
 `migration_lookup` process plugin.
 
-## Handling another section
+### Handling a new section
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 18]
```diff
@@ -115,18 +176,19 @@ handled by the `component` migration. The file
...
 
 Each section of the annual report has a corresponding section of the XML file,
 and we should be able to handle them all as described in the previous section.
 There are some nested Paragraphs, and handling those will be a little
-different.
+different (see `Special Sections` below).
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: > Special Sections

Should be Special sections (sentence casing)

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 19]
```diff
@@ -266,3 +328,111 @@ something like this:
 
 This field expects a `target_id` and a `target_revision_id`, and the
 `combined` value is ignored.
+
+#### Adding new section to batch process
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 20]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+      //...
+      // foia annual report import migration
+      'foia_agency_report',
+    ];
+
+    return $migrations_list;
+  }
+```
+
+### Special Sections
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before

Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: Change to sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 21]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+Note that for certain sections, a sub-paragraph is implemented to capture data 
+provided within the section.
+
+Each case will be different so for examples look at the migrations for 
+section IV where:
+- sub-import: foia_iv_details
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Change to asterisk bullets.
[space][asterisk][space][text]

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 22]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+- main-import: foia_iv_statute
+
+or section V.B.2 where:
+- sub-import: foia_vb2_other
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Change to asterisk bullets.
[space][asterisk][space][text]

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 23]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+- main-import: foia_vb2
+
+TROUBLESHOOTING
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 24]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+---------------
+# Import data from agency annual reports in NIEM-XML format
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 25]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+In general, updating or adding a new migration can be tested using the admin UI.
+
+After making changes in the `config/install/` directory, the usual workflow is
+```
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before this

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 26]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+implemented by the Migrate and Migrate Tools modules.
+
+When testing, you can repeat the first two steps as needed and only export
+configuration when you are ready to commit.
+
+This workflow creates two nearly identical copies of the configuration files:
+on in the module's `config/install/` directory and one in the site's
+configuration directory. There are at least two advantages to this redundancy:
+
+1. We can add comments to the files in the module. Comments are stripped from
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add space before numbered bullet

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 27]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+   the files generated by `drush cex`.
+2. It provides a safeguard in case someone else deletes files in the site
+   configuration directory.
+
+## Running the migrations
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 28]
```diff
@@ -266,3 +328,111 @@ something like this:
...
+after uploading the file.
+
+The other option is to use the command line:
+
+```
+drush @foia.local ms; drush @foia.local mim -vvv --debug component; drush @foia.local mim -vvv --debug --group=foia_component_data_mapping; drush @foia.local mim -vvv --debug --group=foia_component_data_import_subs; drush @foia.local mim -vvv --debug --group=foia_component_data_import; drush @foia.local mim -vvv --debug foia_agency_report --update; drush @foia.local ms
+```
+
+This combines the running of the following import commands with the checking of the migration status before and after:
+```
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: Add line break before this.

File: docroot/modules/custom/foia_upload_xml/README.md [hunk 29]
```diff
@@ -1,61 +1,121 @@
...
+The module creates a new Annual Report node per agency and report year or 
+updates an already existing report for the corresponding agency and report year.
+
+
+REQUIREMENTS
+------------
+This module requires the core Migrate module along with the following contrib 
+modules:
+ * Migrate Plus (https://www.drupal.org/project/migrate_plus)
+ * Migrate Tools (https://www.drupal.org/project/migrate_tools)
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/README.md]: I thought this module also depended on custom module `foia_migrate`. I am not positive though, so just double check to see.

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 4 above)
Comment (bemarlan) [reply docroot/modules/custom/foia_upload_xml/README.md]: Update this wording for consistency:

FOIA Upload XML is a custom Drupal module so unlike contrib modules, the codebase is not
installed via composer. Enable as you would normally enable a contributed Drupal
module. Visit
https://www.drupal.org/docs/8/extending-drupal-8/installing-drupal-8-modules#s-step-2-enable-the-module
for further information.

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 1 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * whitespace added
* Heading level changed

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 2 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 3 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 4 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 5 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 6 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: updated heading levels

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 7 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Added line breaks
* Updated heading level

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 8 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Added line break
* Updated heading level

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 9 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Added line break
* Updated heading level

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 10 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Added line break
* Changed to sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 11 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Added line break
* Changed to sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 12 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Update heading level

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 14 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 15 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Line break added
* Changed to sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 13 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Changed to use asterisk

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 16 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Line break added
* changed to sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 17 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 18 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Changed to sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 19 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 20 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: * Line break added
* changed to sentence casing

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 21 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Changed list syntax to use asterisk

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 22 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Changed list syntax to use asterisk

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 23 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 24 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 25 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 27 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 28 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Line break added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 26 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Spaces added

File: docroot/modules/custom/foia_upload_xml/README.md (hunk 29 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/README.md]: Added foia_migrate as a recommended module.

**Comment by bemarlan** [comment]
Closing in favor of PR 326 https://github.com/acquia-pso/foia-api-enhancements/pull/326

**Pull request closed without merging**